opencv/cvat
# [Datumaro] Dataset annotations filter

Added:
- Dataset annotations filter in CLI: `datumaro project extract -a`

To test:
```
  -e FILTER, --filter FILTER 
                        XML XPath filter expression for dataset items.
                        Examples: 
                        - extract images with width < height:
                        '/item[image/width < image/height]'; 
                        - extract images with large-area bboxes: 
                        '/item[annotation/type="bbox" and annotation/area>2000]'
                        - filter out irrelevant annotations from items: 
                        '/item/annotation[label = "person"]'
  -a, --filter-annotations 
                        Filter annotations instead of dataset items (default: False)
  --remove-empty          
                         Remove an item if there are no annotations left after filtration
  --dry-run         Print XML representations to be filtered and exit

datum project extract -a -e '/item/annotation[label = "person"]' --remove-empty -d temp
```

In general, to filter annotations make an XPath that returns `annotation` XML elements. Check `dry-run` option to obtain XML representations of dataset items.

Resolves #994
